api/v1/logicalvolume_types.go

```diff
@@ -18,6 +35,10 @@ type LogicalVolumeSpec struct {
 	NodeName    string            `json:"nodeName"`
 	Size        resource.Quantity `json:"size"`
 	DeviceClass string            `json:"deviceClass,omitempty"`
+	// +kubebuilder:validation:Optional
+	Snapshot SnapshotSpec `json:"snapshot,omitempty"`
```

Comment:
It is different from the proposal, the following struct looks better for implementation. Could you please consider this?

```suggestion
        // `thin-snapshot`, `thick-snapshot` or empty (not snapshot volume).
	Type string `json:"type"`
	// LogicalVolume name is better than volumeID for expandability (remote copy).
	// Or empty when no need to data copy.
	DataSource string `json:"dataSource"`
	// `AccessType` looks not needed because it is decided by the following rules.
	// ro if Type =~ ".*-snapshot"
	// rw if Type !~ ".*-snapshot" && DataSource != ""
```

Reply:
Thanks for the suggestion @llamerada-jp !

The current implementation is aligned to the proposal, where any snapshot logical volume is mentioned with a new `Snapshot` parameter:
```
Spec: topolvmv1.LogicalVolumeSpec{
	        Name:        sname,
		NodeName:    sourceVol.Spec.NodeName,
		DeviceClass: sourceVol.Spec.DeviceClass,
               // optional; set only in case of snapshots
		Snapshot: topolvmv1.SnapshotSpec{
			Type:       snapType,
			DataSource: sourceVol.Status.VolumeID,
			AccessType: accessType,
		},
	},
```

Here are some of the advantages of keeping a new `Snapshot` field.:

- In case a logical volume is NOT a snapshot (which will mostly be the case), there will be no change in the current LogicalVolume CR spec fields. It will be identical to how it looks now (No new fields!)
For e.g, if a logical volume is not a snapshot, the spec will still look like this:
```
Spec: topolvmv1.LogicalVolumeSpec{
				Name:        name,
				NodeName:    node,
				DeviceClass: dc,
				Size:        *resource.NewQuantity(requestGb<<30, resource.BinarySI),
			},
``` 

- Since this is the initial snapshot effort, more snapshot features to come in in the future. Having a separate `Snapshot` field with snapshot-specific sub-fields makes it easier to add/remove snapshot-related parameters and keep the main LogicalVolume spec untouched.

- Looks clean as it is easier to identify if a logical volume getting created is a snapshot or just a regular logical volume creation via just seeing a `Snapshot` parameter.

What do you think?

Reply:
@llamerada-jp Updated the API as per your suggestions :+1: